subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultIncludedBuild.java

```diff
@@ -81,6 +98,29 @@ public void dependencySubstitution(Action<? super DependencySubstitutions> actio
         return dependencySubstitutionActions;
     }
 
+    @Override
+    public Set<Pair<ModuleVersionIdentifier, ProjectComponentIdentifier>> getAvailableModules() {
+        // TODO: Synchronization
+        if (availableModules==null) {
+            Gradle gradle = getConfiguredBuild();
+            availableModules = Sets.newLinkedHashSet();
+            for (Project project : gradle.getRootProject().getAllprojects()) {
+                registerProject(availableModules, (ProjectInternal) project);
+            }
+        }
+        return availableModules;
+    }
+
+    private void registerProject(Set<Pair<ModuleVersionIdentifier, ProjectComponentIdentifier>> availableModules, ProjectInternal project) {
```

Comment:
nit-pick: this would probably be cleaner if it just used the `availableModules` field rather than taking it as a parameter.